c2rust-analyze/src/util.rs

```diff
@@ -145,11 +149,92 @@ pub fn ty_callee<'tcx>(tcx: TyCtxt<'tcx>, ty: Ty<'tcx>) -> Option<Callee<'tcx>>
     })
 }
 
+trait IsTrivial {
+    /// Something [`is_trivial`] if it has no effect on pointer permissions,
+    /// and thus requires no special handling.
+    /// That is, it must contain no raw pointers.
+    ///
+    /// See [`Trivial`] for more info.
+    ///
+    /// [`is_trivial`]: Self::is_trivial
+    /// [`Trivial`]: Callee::Trivial
+    fn is_trivial(&self) -> bool;
+}
+
+impl IsTrivial for Ty<'_> {
+    /// A [`Ty`] [`is_trivial`] if
+    /// it is not a (raw) pointer and contains no (raw) pointers.
+    ///
+    /// [`is_trivial`]: IsTrivial::is_trivial
+    fn is_trivial(&self) -> bool {
+        // Similar logic to [`Ty::contains`].
+        struct Visitor;
+
+        impl<'tcx> TypeVisitor<'tcx> for Visitor {
+            type BreakTy = ();
+
+            fn visit_ty(&mut self, ty: Ty<'tcx>) -> ControlFlow<Self::BreakTy> {
```

Comment:
using `ControlFlow` feels like overkill. It's pretty obvious what returning `true` or `false` would mean here in this visitor, and no value in `Break(..)` or `Continue(..)` is used anyway.

Reply:
This is implementing a trait method from rustc, and [the trait declaration](https://doc.rust-lang.org/stable/nightly-rustc/rustc_middle/ty/visit/trait.TypeVisitor.html) uses `ControlFlow`.